airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryUtils.java

```diff
@@ -162,6 +166,27 @@ public static JsonNode getGcsAvroJsonNodeConfig(final JsonNode config) {
     return gcsJsonNode;
   }
 
+  public static String getDatasetId(final JsonNode config) {
+    String datasetId = config.get(BigQueryConsts.CONFIG_DATASET_ID).asText();
+    Matcher matcher = DATASET_ID_PATTERN.matcher(datasetId);
```

Comment:
I think the regex makes it a little hard to follow the logic; string manipulation feels more clear:
```java
int colonIndex = datasetId.indexOf(":");
if (colonIndex != -1) {
  String projectId = datasetId.substring(0, colonIndex);
  // check equality against config.get(PROJECT_ID)
  // ...
}
// if colonIndex is -1, then this returns the entire string
// otherwise it returns everything after the colon
return datasetId.substring(colonIndex + 1);
```
This means we would lose the validation aspect (e.g. if someone enters `invalid*characters!` we'll accept it). But those errors would be detected when the connector runs its `check` method, so that feels OK to me

Reply:
I see, I agree with your suggestion. We don't need to check here what BigQuery checks.